src/semidiscretization/semidiscretization.jl

```diff
@@ -62,8 +62,6 @@ Wrap the semidiscretization `semi` as an ODE problem in the time interval `tspan
 that can be passed to `solve` from the [SciML ecosystem](https://diffeq.sciml.ai/latest/).
 """
 function semidiscretize(semi::AbstractSemidiscretization, tspan)
-  RealT = real(semi)
-  tspan = (convert(RealT, first(tspan)), convert(RealT, last(tspan)))
```

Comment:
Why did you sneakily remove these lines? They allow to quickly change the `tspan` from `trixi_include`. Otherwise, IIRC, there are subtle errors if you pass `(0, 0.1)` as the time span.

I know you strongly dislike unclean typing, but shouldn't we then maybe at least produce a useful error message? I.e., via a dispatch on `semidiscretize` with `tspan::Union{Tuple{Int, Real}, Tuple{Real, Int}, Tuple{Int, Int}}` or something similar?

Reply:
You can still pass `tspan = (0.0, 1)` since it's normalized in OrdinaryDiffEq. You're not allowed to pass `tspan = (0, 1)`. On the other hand, this change allows users to easily change the time type, e.g. when differentiating through a solver with a stepsize callback where the time needs to be a dual number. 
Using a very ugly dispatch for such user errors is not something I like very much... In contrast, I would like to educate people about Julia's type system instead of hiding it. People should know that `1` is not the same as `1.0`.

Reply:
Aaaaaarrrrrrggggghhhhh...